sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/BlobEventStoreEventSource.cs

```diff
@@ -439,15 +441,173 @@ protected BlobEventStoreEventSource() : base(EventSourceName)
         ///
         [Event(38, Level = EventLevel.Error, Message = "An exception occurred when retrieving checkpoint for FullyQualifiedNamespace: '{0}'; EventHubName: '{1}'; ConsumerGroup: '{2}'; PartitionId: '{3}'; ErrorMessage: '{4}'.")]
         public virtual void GetCheckpointError(string fullyQualifiedNamespace,
-                                                 string eventHubName,
-                                                 string consumerGroup,
-                                                 string partitionId,
-                                                 string errorMessage)
+                                               string eventHubName,
+                                               string consumerGroup,
+                                               string partitionId,
+                                               string errorMessage)
         {
             if (IsEnabled())
             {
                 WriteEvent(38, fullyQualifiedNamespace ?? string.Empty, eventHubName ?? string.Empty, consumerGroup ?? string.Empty, partitionId ?? string.Empty, errorMessage ?? string.Empty);
             }
         }
+
+        /// <summary>
+        /// Writes an event with three string arguments and a value type into a stack allocated
+        /// <see cref="EventData"/> struct to avoid the parameter array allocation on the WriteEvent methods.
+        /// </summary>
+        /// <param name="eventId">The event identifier.</param>
+        /// <param name="arg1">The first argument.</param>
+        /// <param name="arg2">The second argument.</param>
+        /// <param name="arg3">The third argument.</param>
+        /// <param name="arg4">The fourth argument.</param>
+        [NonEvent]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe void WriteEvent<TValue1>(int eventId, string arg1, string arg2, string arg3, TValue1 arg4)
```

Comment:
I wonder... do we have enough overlap with some of these non-specific ones that we should consider a set of `internal` extension methods that we share and can call from any of the Event Hubs sources?